File: blinkvoice-visual-sdk/src/main/java/com/blinkvoice/visual/api/BlinkCaptureResult.java
package com.blinkvoice.visual.api;

public final class BlinkCaptureResult {
    private final BlinkEventType eventType;
    private final long startTimeMs;
    private final long endTimeMs;
    private final long durationMs;
    private final float confidence;

    public BlinkCaptureResult(
            BlinkEventType eventType,
            long startTimeMs,
            long endTimeMs,
            long durationMs,
            float confidence
    ) {
        this.eventType = eventType;
        this.startTimeMs = startTimeMs;
        this.endTimeMs = endTimeMs;
        this.durationMs = durationMs;
        this.confidence = confidence;
    }

    public BlinkEventType getEventType() {
        return eventType;
    }

    public long getStartTimeMs() {
        return startTimeMs;
    }

    public long getEndTimeMs() {
        return endTimeMs;
    }

    public long getDurationMs() {
        return durationMs;
    }

    public float getConfidence() {
        return confidence;
    }
}

File: IosFloat/ios-float/BlinkVoiceVisual/BlinkCaptureResult.swift
public struct BlinkCaptureResult: Codable, Equatable, Sendable {
    public let eventType: BlinkEventType
    public let startTimeMs: Int64
    public let endTimeMs: Int64
    public let durationMs: Int64
    public let confidence: Float

    public init(
        eventType: BlinkEventType,
        startTimeMs: Int64,
        endTimeMs: Int64,
        durationMs: Int64,
        confidence: Float
    ) {
        self.eventType = eventType
        self.startTimeMs = startTimeMs
        self.endTimeMs = endTimeMs
        self.durationMs = durationMs
        self.confidence = confidence
    }
}
